Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/SevenZipExtractor.java

```diff
@@ -789,138 +787,100 @@ boolean unpack(AbstractFile archiveFile, ConcurrentHashMap<Long, Archive> depthM
     }
 
     /**
-     * Stream used to unpack the archive to local file
+     * Stream used to unpack the archive item to local file. This will be used when 
+     * the 7ZIP bindings call getStream() on the StandardIArchiveExtractCallback.
      */
-    private abstract static class UnpackStream implements ISequentialOutStream {
-
-        private OutputStream output;
-        private String localAbsPath;
-
-        UnpackStream(String localAbsPath) {
-            this.localAbsPath = localAbsPath;
-            try {
-                output = new EncodedFileOutputStream(new FileOutputStream(localAbsPath), TskData.EncodingType.XOR1);
-            } catch (IOException ex) {
-                logger.log(Level.SEVERE, "Error writing extracted file: " + localAbsPath, ex); //NON-NLS
-            }
-
-        }
-
-        public abstract long getSize();
+    private final static class MutableEncodedFileOutputStream extends EncodedFileOutputStream 
+            implements AutoCloseable {
+        
+        private static final int HEADER_LENGTH = 32;
+        private static final String HEADER = "TSK_CONTAINER_XOR1_xxxxxxxxxxxxx";
```

Comment:
While we can't change them at this point, having the XOR1 header and encoding method hard-coded here seems wrong. But they also don't seem like the kind of thing we want in the public API. I wonder if it would make more sense to create a public MutableEncodedFileOutputStream class over in the same package as EncodedFileOutputStream since the getHeader and encodeByte methods are package scope.

Reply:
@rcordovano your suggestion has now been implemented. I ran a test archive to verify that it is functioning correctly. Its now a candidate for final review by @APriestman and integration.